### PR TITLE
[sensors-glib] Initial implementation. Fixes JB#61103

### PR DESCRIPTION
**Pull request by spiiroin**
Provides C language client interface for communicating with Sailfish Sensor Framework daemon via a set of glib objects.

### CODE REVIEW

File: sfwreporting.c [hunk 1]
```diff
@@ -0,0 +1,1184 @@
...
+    int                rpt_override_requested;
+    int                rpt_override_effective;
+    GCancellable      *rpt_override_cancellable;
+
+} SfwReportingPrivate;
+
+struct SfwReporting
+{
+    GObject object;
+};
```
Comment (monich) [inline sfwreporting.c:115]: Split into public and private parts only make sense if one part is public and the other one is private. If the whole thing is private (defined in the .c file and not accessible from elsewhere) then why not to define a single structure.

File: sfwreporting.h [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+/* ========================================================================= *
+ * Prototypes
+ * ========================================================================= */
+
+/* ------------------------------------------------------------------------- *
+ * SFWREPORTING_TYPE
+ * ------------------------------------------------------------------------- */
+
+GType sfwreporting_get_type(void);
```
Comment (monich) [inline sfwreporting.h]: If `SfwReporting` structure (and its inheritance from `GObject`) is private then there's no need to expose these type getters in the API

Comment (spiiroin) [reply sfwreporting.h]: mmm, yes. slipped through. thanks!

File: sfwreporting.c (hunk 1 above)
Comment (spiiroin) [reply sfwreporting.c:115]: IIRC there was some guidance along the line: using public/private data separation can make it easier to make objects inheritable later on or stte.

Comment (monich) [reply sfwreporting.c:115]: If there are plans to expose the object and especially make it inheritable - then yes, it does make sense.

**Comment by spiiroin** [comment]
Review commits squashed

**Comment by LaakkonenJussi** [comment]
Why not have all sources in `/src` and headers to include in `/include`? Would be quite obvious what one would need to look at when using this.

**Comment by spiiroin** [comment]
> Why not have all sources in `/src` and headers to include in `/include`? Would be quite obvious what one would need to look at when using this.

I'm not sure if I follow. Having sources and headers in different directories never made *any* sense to me. Having sources in separate dir(s) is okay, but I usually do it only if there sources with different purposes i.e. having "daemon" and "client" = sure, "src" = not so sure it is useful. Though there is "examples" dir, so maybe "src", or "library", or something would be justifiable?

**Comment by monich** [comment]
I guess Jussi means clear separation between interface and implementation. It does make sense to me. Interfaces are public and stable, internal details and implementation are subject to change.

It doesn't make that much sense though, if every header is public. In that case there isn't much to separate - it's clear enough that .h files are the interfaces, .c files contain the implementation.

**Review by rainemak — APPROVED** [review]
LGTM

**Pull request merged**